src/SimpleApi/Startup.cs

```diff
@@ -0,0 +1,52 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using MongrationDotNet;
+
+namespace SimpleApi
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddControllers();
+
+            var mongoDbConfig = Configuration.GetSection("MongoDB");
+            var mongoOptions = mongoDbConfig.Get<MongoDbOptions>() ?? new MongoDbOptions();
+
+            services.AddMigration(mongoOptions.Server, mongoOptions.DatabaseName)
+                .WithAllAvailableMigrations();
+
+            services.AddTransient<IStartupTask, SetupMongoMigration>();
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseHttpsRedirection();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints => { endpoints.MapControllers(); });
+
+            app.RunStartupTasks();
+        }
```

Comment:
Maybe even `app.RunMongration();`

Reply:
This code is for running all startup tasks as users may have more startup tasks.